server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java

```diff
@@ -506,8 +473,8 @@ public void testCheckAndReportError() {
 
         var version = new ReservedStateVersion(2L, Version.CURRENT);
         var error = controller.checkAndReportError("test", List.of("test error"), version);
-        assertThat(error, allOf(notNullValue(), instanceOf(IllegalStateException.class)));
-        assertEquals("Error processing state change request for test, errors: test error", error.getMessage());
+        assertThat(error, instanceOf(IllegalStateException.class));
+        assertThat(error.getMessage(), is("Error processing state change request for test, errors: test error"));
```

Comment:
Again I wonder why this is preferred? For uniformity with the other asserts in the same test?

Reply:
Exactly